File: src/main/java/br/com/objective/gourmetgame/EmptyResponse.java
package br.com.objective.gourmetgame;

import javax.swing.*;

public class EmptyResponse implements Question {

    public static final String TEXT_QUESTION_NEW = "Qual prato que você pensou?";
    public static final String TEXT_QUESTION_IS = "%s é ___________ mas %s não.";
    private static final String ACERTEI_DE_NOVO = "Acertei de novo!";

    @Override
    public void quiz() {
        throw new UnsupportedOperationException("Não há informações do prato para chamada deste método");
    }

    @Override
    public void positive() {
        JOptionPane.showMessageDialog(null, ACERTEI_DE_NOVO);
    }

    @Override
    public void negative(Question question) {
        String otherDish = JOptionPane.showInputDialog(null, TEXT_QUESTION_NEW, titleQuestion, JOptionPane.QUESTION_MESSAGE);
        String txtQuestion = String.format(TEXT_QUESTION_IS, otherDish, question.getInformation());
        String difference = JOptionPane.showInputDialog(null, txtQuestion, titleQuestion, JOptionPane.QUESTION_MESSAGE);
        question.addQuestion(otherDish,difference);
    }

    @Override
    public void addQuestion(String otherDish,String difference) {
        throw new UnsupportedOperationException("Não há informações do prato para inclusão da questão");
    }

    @Override
    public String getInformation() {
        throw new UnsupportedOperationException("Não há informações do prato para obter as informações");
    }

}

File: src/app/model/emptyResponse.ts
import { Question } from "src/app/model/question";
import { MatDialog } from "@angular/material";
import { DialogInfoComponent } from "src/app/dialog-info/dialog-info.component";
import { ReflectiveInjector } from "@angular/core";
import { Inject } from "@angular/core";
import { DialogInputComponent } from "src/app/dialog-input/dialog-input.component";


export class EmptyResponse implements Question {
    
    private TEXT_QUESTION_NEW:string = "Qual prato que você pensou ?";
    private ACERTEI_DE_NOVO:string = "Acertei de novo!";
    constructor(private dialog:MatDialog){        }

    quiz() {
        throw new Error("Não há informações do prato para chamada deste método");
    }

    positive() {
        this.dialog.open(DialogInfoComponent, {
            data: {
                message: this.ACERTEI_DE_NOVO
            }
        });
    }

    negative(question: Question) {
        let otherDish, difference;
        this.dialog.open(DialogInputComponent, {
            data: {
                message: this.TEXT_QUESTION_NEW
            }, disableClose : true
        }).afterClosed().subscribe(data=>{
            otherDish = data;
            this.callDiference(question, data);
        });
    }

    callDiference(question:Question, otherDish){
        this.dialog.open(DialogInputComponent, {
            data: {
                message: `${otherDish} é ___________ mas ${question.getInformation()} não.`
            }, disableClose : true
        }).afterClosed().subscribe(data=>{
            question.addQuestion(otherDish, data);
        })
    }

    addQuestion(otherDish: string, difference: string) {
        throw new Error("Não há informações do prato para inclusão da questão");
    }
    getInformation(): string {
        throw new Error("Não há informações do prato para obter as informações");
    }

}